ckanext/ontario_theme/templates/internal/package/resource_validation.html

```diff
@@ -23,19 +28,76 @@
+    <form "method=post" action="{{ action }}">
+      <button id="nextBtn"
+              class="ontario-button ontario-button--primary"
+              name="save"
+              type="submit">{{ _('Continue') }}</button>
+    </form>
   {% endif %}
 
-  {% set package_id = resource.get('package_id') %}
-  {% set resource_id = resource.get('id') %}
-  {% set action = h.url_for('datastore.dictionary',
-      id=package_id,
-      resource_id=resource.id) %}
- 
-  <form method="get" action="{{ action }}">
-    <button class="ontario-button ontario-button--primary"
-            name="save"
-            type="submit">Continue</button>
-  </form>
+  {% if is_csv == "True" %}
+    <script>
+      function poll_datastore() {
+        const resId = "{{ resource.id }}";
+        const xhr = new XMLHttpRequest();
+        const url = `/api/3/action/resource_show?id=${resId}`
+
+        xhr.open("POST", url, true);
+        xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
+        xhr.send();
+
+        xhr.onreadystatechange = () => {                                
+          if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+            let thisResponse = JSON.parse(xhr.response);
+
+            if (thisResponse.result.datastore_active) {
+              const sleep = (delay) => new Promise((resolve) => setTimeout(resolve, delay))
+         
+              const fixedDelay = async () => {
+                await sleep(2000)
+                document.getElementById("pendingTxt").innerHTML = "Data file uploaded. Please click Continue.";
+                document.getElementById("spinnerSVG").style.visibility = "hidden";
+                document.getElementById("step2Btn").style.visibility = "visible";
+              }
+              fixedDelay();
+
+            } else {
+            setTimeout(poll_datastore, 1000);
+            }
+          }
+        }
+      };
+      poll_datastore();
+    </script>
+
+    <div class="ontario-small-12 ontario-row">
+      <div class="ontario-small-12 ontario-columns">
+        <div id="loadingSpinner" aria-hidden="false" role="alert">
+          <div class="ontario-loading-indicator">
+            <svg id="spinnerSVG"
+                 class="ontario-loading-indicator__spinner-small"
+                 viewBox="25 25 50 50"
+                 xmlns="http://www.w3.org/2000/svg">
+              <circle cx="50" cy="50" r="20" fill="none" stroke-width="4" />
+            </svg>
+            <p class="spinner-small" id="pendingTxt">Uploading data file to database</p>
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <form "method=post" action="{{ action }}">
+      <button id="step2Btn"
+              style="visibility: hidden;
+                     margin-top: 40px;
+                     margin-left: 36px"
+              class="ontario-button ontario-button--primary"
+              name="save"
+              type="submit">{{ _('Continue') }}</button>
+    </form>
```

Comment:
There is a misalignment with the button and the text once the spinner is not visible to users. To align the button and text to the left of the screen, remove the margin-left from the button and the text. Instead of using "visibility: hidden;" use "display: none;" in both the button and the javascript. In the javascript, `visibility: visible` would be `display: block`. You'd remove the `margin-left: 12px` from `.ontario-loading-indicator p.spinner-small` and add `margin-right: 12px` to `.ontario-loading-indicator__spinner-small`

Reply:
This will misalign the text after the spinner disappears:
![Screenshot 2024-07-03 at 16 24 10](https://github.com/ongov/ckanext-ontario_theme/assets/1254764/de81f4fb-925c-47ee-b815-82cffb00bbe8)

Reply:
That's because you are using `visibility: hidden;` for the style of the SVG using `display: none;` will fix the alignment.